File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/window/REVerticalWindow.java
package io.ganguo.demo.view.window;

import android.content.Context;

import io.ganguo.demo.viewmodel.window.REVerticalWindowVModel;
import io.ganguo.viewmodel.databinding.IncludeRecyclerPopupwindowBinding;
import io.ganguo.vmodel.ViewModelPopupWindow;
/**
 * Vertical 方向 - RecyclerView PopupWindow
 * Created by leo on 2018/10/21.
 */
public class REVerticalWindow extends ViewModelPopupWindow<IncludeRecyclerPopupwindowBinding, REVerticalWindowVModel> {
    public REVerticalWindow(Context context) {
        super(context);
    }

    @Override
    public void onViewAttached(REVerticalWindowVModel viewModel) {

    }

    @Override
    public boolean isWindowWidthFull() {
        return false;
    }

    @Override
    public boolean isWindowHeightFull() {
        return false;
    }

    @Override
    public boolean isTouchOutsideDismiss() {
        return true;
    }

    @Override
    public REVerticalWindowVModel createViewModel() {
        return new REVerticalWindowVModel();
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/window/REVerticalWindow.kt
package io.ganguo.demo.view.window

import android.content.Context

import io.ganguo.demo.viewmodel.window.REVerticalWindowVModel
import io.ganguo.viewmodel.databinding.IncludeRecyclerPopupwindowBinding
import io.ganguo.viewmodel.core.view.ViewModelWindow

/**
 * Vertical 方向 - RecyclerView PopupWindow
 * Created by leo on 2018/10/21.
 */
class REVerticalWindow(context: Context) : ViewModelWindow<IncludeRecyclerPopupwindowBinding, REVerticalWindowVModel>(context) {


    override fun createViewModel(): REVerticalWindowVModel {
        return REVerticalWindowVModel()
    }

    override fun onViewAttached(viewModel: REVerticalWindowVModel) {
    }

}
